enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,246 @@
+and scaling [recommendations](https://www.elastic.co/blog/how-many-shards-should-i-have-in-my-elasticsearch-cluster).
+
+## Motivation
+
+The initial data design for Cluster Logging segments logs by OpenShift namespace in order to facilitate multi-tenant support and data curation.  This choice was made because
+index level security was the only feature available from an open source library.  It additionally facilitated curation as full indicies could be removed when the retention period
+expired.  This means, however, at any one time, there are at least "**$noOfNamespaces * $daysRetatined**" number 
+of indices maintained by the Elasticsearch server.  Each index can additionally be sharded to spread the load across the Elasticsearch nodes.  The end
+result of this shard explosion is the Elasticsearch cluster performance is not optimized and is not capable of efficiently processing logs or storing as many as it could. 
+
+Each index adds load and overhead (e.g. mapping, metadata) to the Elasticsearch cluster that needs to be tracked beyond the actual data.   Elasticsearch has recommendations for maximum shard size and the number of shards per node per allocated gigabyte of heap.  Cluster logging typically exceeds these recommendations for any OpenShift clusters that has significant log traffic.
+
+
+### Goals
+
+
+The goals of this proposals are:
+* Utilize a data design that aligns data schema with Elasticsearch's recommendations.
+* Expose data management policy as API in the `cluster-logging-operator` and `elasticsearch-operator` in support of Cluster Logging's mission to gather log sources
+* Migrate indicies from the previous schema into the new.  Migrated inidices will be governed by the data management policy exposed by this proposal
+ 
+
+### Non-Goals
+
+This change will not:
+* Provide a general data management policy API that fully exposes Elasticsearch's rollover API
+
+## Proposal
+
+This proposal introduces two specific changes to to achieve its goals:
+
+* Co-located data in a few opinionated indices
+* Index management using [rollover index API](https://www.elastic.co/guide/en/elasticsearch/reference/6.3/indices-rollover-index.html)
+
+Logs of a given type (e.g. app container, infra container, node) are separated by index.  The Cluster Logging collector writes logs to a well-known alias established by the `cluster-logging-operator`.  The ClusterLogging CR instance specifies the management policy for each log type index.  This controls: 
+* the maximum age (e.g. 7 days).  
+
+This policy is passed to the Elasticsearch CR for the `elasticsearch-operator` to manage the rollover policy.  The details of the policy are specified by the `cluster-logging-operator` and are based on the guidelines suggested by Elasticsearch.
+
+
+### Implementation Details
+
+#### Assumptions
+
+* ClusterLogging will expose the minimal needed set of Elasticsearch rollover policy management API in order to achieve the previously described goals
```

Comment:
can we clarify which "Elasticsearch" we're referring to here? I'm assuming its the `elasticsearch` CRD and not the component Elasticsearch...

Reply:
This line "Elasticsearch rollover policy management API" I am referring  to the actual Elasticsearch rollover api.